### PR TITLE
chore(ci): Track released artifacts for EO 14028 compliance.

### PR DESCRIPTION
**Pull request by hessjcg**
## Change Description

The release job will now build the package .tgz and report it to Kokoro using `define_artifacts` so that
Kokoro can generate an SBOM for this artifact.

### CODE REVIEW

File: .kokoro/publish.sh [hunk 1]
```diff
@@ -28,4 +28,24 @@ NPM_TOKEN=$(cat $KOKORO_KEYSTORE_DIR/73713_google-cloud-npm-token-1)
...
+
+npm pack .
+
+# Second, publish the tarball that was just created
+
+# npm provides no way to specify, observe, or predict the name of the tarball
+# file it generates.  We have to look in the current directory for the freshest
+# .tgz file.
+TARBALL=$(ls -1 -t *.tgz | head -1)
+npm publish --access=public --registry=https://wombat-dressing-room.appspot.com "$TARBALL"
```
Comment (ruyadorno) [inline .kokoro/publish.sh:42]: `npm pack` actually output the file name to stdout, so this should be feasible with something like: 
```suggestion

# First, pack this project as a tarball

TARBALL_BASENAME=$(npm pack) .

# Second, publish the tarball that was just created

TARBALL="$(pwd)/$TARBALL_BASENAME"
npm publish --access=public --registry=https://wombat-dressing-room.appspot.com "$TARBALL"
```

**Comment by ruyadorno** [comment]
we should change the commit prefix to `chore(ci):` instead of `ci:` (can be done at the moment of squash-merge if you prefer) 😊

**Comment by hessjcg** [comment]
I accepted your suggestion. I'll share it with the other folks doing node projects.

**Comment by ruyadorno** [comment]
@hessjcg awesome, it'd be nice if we could run some sort of end-to-end test and ensure that it's working as intended ahead of the next release

**Review by ruyadorno — APPROVED** [review]

**Pull request merged**